File: MmoGameUnity/Assets/Game/Hoftix/Script/GenerateScripts/UIModules/LoadingPanelView.cs
﻿using ZJYFrameWork.UISerializable.UIInitView;
using UnityEngine;
using UnityEngine.UI;
using ZJYFrameWork.Hotfix.UISerializable;

namespace ZJYFrameWork.UISerializable
{
    public class LoadingPanelView:UIViewInterface
    {
        public UnityEngine.UI.Slider leftSlider_Slider=null;
		public UnityEngine.UI.Slider rightSlider_Slider=null;
		public UnityEngine.UI.Text progressNum_Text=null;
		public LoadUIController LoadingController=null;
		


        public void Init(UIView _view)
        {
            leftSlider_Slider=_view.GetObjType<UnityEngine.UI.Slider>("leftSlider_Slider");
			rightSlider_Slider=_view.GetObjType<UnityEngine.UI.Slider>("rightSlider_Slider");
			progressNum_Text=_view.GetObjType<UnityEngine.UI.Text>("progressNum_Text");
			LoadingController=_view.GetObjType<LoadUIController>("LoadingController");
			
        }
    }
}

File: DemoMMOUnity/Lua/application/app/ui/generateScripts/UIModules/LoadingPanelView.lua
---@class LoadingPanelView
LoadingPanelView = class("LoadingPanelView")
function LoadingPanelView:Init(view)
	self._UIView = view:GetComponent("UIView")
	self.leftSlider_Slider = self._UIView:GetObjType("leftSlider_Slider") or UnityEngine.UI.Slider
	self.rightSlider_Slider = self._UIView:GetObjType("rightSlider_Slider") or UnityEngine.UI.Slider
	self.progressNum_Text = self._UIView:GetObjType("progressNum_Text") or UnityEngine.UI.Text
	self.LoadingController = self._UIView:GetObjType("LoadingController") or UnityEngine.Object
end

